### PR TITLE
Complete requirements for the framework

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

**Comment by aeslaughter** [comment]
@permcody I made some mistake with a rebase, I will fix it next week.

**Comment by aeslaughter** [comment]
```
test/tests
                   Complete: 100.0%
      Total Number of Files: 720
    Files with Requirements: 0
      Total Number of Tests: 1873
    Tests with Requirements: 1784
           Deprecated Tests: 89
            Tests Remaining: 0
```

File: test/tests/variables/block_aux_kernel/tests [hunk 1]
```diff
@@ -1,8 +1,13 @@
 [Tests]
+  design = 'syntax/AuxVariables/index.md'
+  issues = ''
```
Comment (permcody) [inline test/tests/variables/block_aux_kernel/tests]: Odd - from here on down you supplied no issue. I thought MooseDocs didn't allow this anymore?

File: test/tests/postprocessors/vector_postprocessor_comparison/tests [hunk 2]
```diff
@@ -1,38 +1,45 @@
...
+      type = 'CSVDiff'
+      input = 'vector_postprocessor_comparison.i'
+      cli_args = 'Functions/b_fn/type=ConstantFunction Functions/b_fn/value=2 '
+                 'Postprocessors/vpp_comparison/comparison_type=equals Outputs/file_base=equals'
+      csvdiff = 'equals.csv'
+
+      detail = "equal."
+    []
+  []
+[]
```
Comment (permcody) [inline test/tests/postprocessors/vector_postprocessor_comparison/tests]: It looks like you have a lot of files without newlines now. It'd be nice to fix these up.

**Review by permcody — CHANGES_REQUESTED** [review]
Let's get issues into those last few tests. If you look at them in the diff view they all seem to be clustered at the bottom.

**Review by permcody — DISMISSED** [review]

**Review by permcody — APPROVED** [review]

**Pull request merged**